lib/msf/core/post/file.rb

```diff
@@ -859,14 +879,22 @@ def _win_ansi_append_file(file_name, data, chunk_size = 5000)
     write_length = [chunk_size, data.length].min
     while start_index < data.length
       begin
-        session.shell_command_token("<nul set /p=\"#{data[start_index, write_length]}\" >> \"#{file_name}\"")
+        success = _shell_command_with_success_code("echo | set /p x=\"#{data[start_index, write_length]}\">> \"#{file_name}\"")
+        unless success
+          vprint_status("Write partially succeeded then failed. May need to manually clean up") unless start_index == 0
+          return false
+        end
         start_index += write_length
         write_length = [chunk_size, data.length - start_index].min
       rescue ::Exception => e
         print_error("Exception while running #{__method__}: #{e}")
+        vprint_status("May need to manually clean up") unless start_index == 0
```

Comment:
We might need to show this by default if it's important to cleanup left behind files 👀

Reply:
Good call - done these.